### ISSUE TITLE
Replace template references to settings.BASE_URL with template tag call

### ISSUE THREAD

**Issue by thenewguy** [wagtail/wagtail#5719]
A hardcoded string reference to the settings.BASE_URL isn't very flexible and is problematic for cleanly serving two domains from one running codebase.

There is a template tag [base_url_setting()](https://github.com/wagtail/wagtail/blob/master/wagtail/admin/templatetags/wagtailadmin_tags.py#L199) that could be used instead.

It would also be nice if this template tag would accept a callable for a dynamic value.

**Comment by gasman** [comment]
Hi @thenewguy - can you clarify which particular templates you're referring to?

**Comment by thenewguy** [comment]
@gasman I hit the issue when attempting to preview page changes.  Then I did a search against the project code for `settings.BASE_URL`.  Unfortunately, I did not keep the list.  After letting this roll around in the back of my head for a few days, I am leaning towards using a custom object for  `settings.BASE_URL` that converts to the appropriate string value.  It won't have access to the request, which is probably generally useful, but I think I can get around that for my needs
